app/src/main/java/com/example/roomies/SettingsFragment.java

```diff
@@ -180,8 +195,40 @@ public void done(List<UserCircle> userCircles, ParseException e) {
                 }
 
                 // save received posts to list and notify adapter of new data
+                userCircle = userCircles.get(0);
                 circle = userCircles.get(0).getCircle();
             }
         });
     }
+
+    // delete userCircle object connection current user and their current circle
+    public void leaveCircle(){
+        // get userCircle object
+        getCircle();
+
+        ParseQuery<ParseObject> query = ParseQuery.getQuery("UserCircle");
+
+        // Retrieve the object by id
+        query.getInBackground(userCircle.getObjectId(), (object, e) -> {
+            if (e == null) {
+                //Object was fetched
+                //Deletes the fetched ParseObject from the database
+                object.deleteInBackground(e2 -> {
+                    if(e2==null){
+                        Toast.makeText(getActivity(), "Left circle", Toast.LENGTH_SHORT).show();
```

Comment:
Can we update the message to be more specific: e.g., "You have left circle XXX"?